src/index/Index.cpp

```diff
@@ -315,9 +323,41 @@ void Index::createPermutation(ExtVec* vec,
     LOG(INFO) << "Size after: " << vec->size() << std::endl;
   }
 
-  createPermutationImpl<MetaData>(_onDiskBase + ".index" + p._fileSuffix, *vec,
-                                  p._keyOrder[0], p._keyOrder[1],
-                                  p._keyOrder[2]);
+  return createPermutationImpl<MetaData>(_onDiskBase + ".index" + p._fileSuffix,
+                                         *vec, p._keyOrder[0], p._keyOrder[1],
+                                         p._keyOrder[2]);
+}
+
+// ________________________________________________________________________
+template <class MetaData, class Comparator1, class Comparator2>
+void Index::createPermutationPair(ExtVec* vec,
+                                  Permutation::PermutationImpl<Comparator1> p1,
```

Comment:
As below I'd pass the `PermutationImpl` object as const references. I realize it's not a big problem as this is called only a hand full of times though.